kyaml/kio/byteio_writer.go

```diff
@@ -118,20 +133,24 @@ func (w ByteWriter) Write(nodes []*yaml.RNode) error {
 	for i := range nodes {
 		items.Content = append(items.Content, nodes[i].YNode())
 	}
-	err := w.encode(encoder, doc)
+	err := w.encode(encoder, false, doc)
 	yaml.UndoSerializationHacksOnNodes(nodes)
 	return err
 }
 
 // encode encodes the input document node to appropriate node format
-func (w ByteWriter) encode(encoder *yaml.Encoder, doc *yaml.Node) error {
+func (w ByteWriter) encode(encoder *yaml.Encoder, forceJSON bool, doc *yaml.Node) error {
 	rNode := &yaml.RNode{}
 	rNode.SetYNode(doc)
-	str, err := rNode.String()
-	if err != nil {
-		return errors.Wrap(err)
+	useJSONEncoder := forceJSON
+	if !forceJSON {
+		str, err := rNode.String()
+		if err != nil {
+			return errors.Wrap(err)
+		}
+		useJSONEncoder = json.Valid([]byte(str))
```

Comment:
I'm not sure if this stuff is still relevant. Maybe it's useful for getting more nicely formatted JSON compatible output when there is no path annotation?